Remove events deprecated field

wojtek-t opened the pull request:
```release-note
The unused `series.state` field, deprecated since v1.14, is removed from the `events.k8s.io/v1beta1` and `v1` Event types.
```

/kind cleanup

wojtek-t committed:
Remove deprecated State fiel in EventSeries

wojtek-t committed:
Autogenerated

wojtek-t commented:
/retest

wojtek-t commented:
/retest

wojtek-t commented on staging/src/k8s.io/api/core/v1/types.go:
```diff
@@ -5215,19 +5215,8 @@ type EventSeries struct {
 	LastObservedTime metav1.MicroTime `json:"lastObservedTime,omitempty" protobuf:"bytes,2,name=lastObservedTime"`
-	// State of this Series: Ongoing or Finished
-	// Deprecated. Planned removal for 1.18
-	State EventSeriesState `json:"state,omitempty" protobuf:"bytes,3,name=state"`
```
@liggitt - the test failures are relevant to this change. Does it mean we can't remove those fields?
Or should I fix the tests?

liggitt replied:
per https://github.com/kubernetes/kubernetes/pull/75987#issuecomment-479309114, go ahead and update the tests (run with `UPDATE_COMPATIBILITY_FIXTURE_DATA=true`) and capture the test output to include as a diff

liggitt commented on staging/src/k8s.io/api/core/v1/types.go:
```diff
@@ -5215,19 +5215,8 @@ type EventSeries struct {
 	LastObservedTime metav1.MicroTime `json:"lastObservedTime,omitempty" protobuf:"bytes,2,name=lastObservedTime"`
-	// State of this Series: Ongoing or Finished
-	// Deprecated. Planned removal for 1.18
-	State EventSeriesState `json:"state,omitempty" protobuf:"bytes,3,name=state"`
```
tombstone the field so we don't reuse this proto or json tag. examples:

https://github.com/kubernetes/kubernetes/blob/e97c570a4ba5ba1e2285d3278396937feaa15385/staging/src/k8s.io/api/core/v1/types.go#L4311-L4313

https://github.com/kubernetes/kubernetes/blob/e97c570a4ba5ba1e2285d3278396937feaa15385/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1118-L1119

wojtek-t replied:
Thanks for catching.

liggitt commented on staging/src/k8s.io/api/testdata/v1.18.0/events.k8s.io.v1beta1.Event.yaml:
```diff
@@ -60,5 +60,4 @@ reportingInstance: "20"
 series:
   count: -1971381490
   lastObservedTime: "2429-02-18T17:57:56.343118Z"
-  state: 鯹)晿<
```
actually commit the .after_roundtrip files, don't modify the originals...

wojtek-t replied:
Done - sorry, didn't realize we should commit those.

wojtek-t committed:
Update compatibility test

--- FAIL: TestCompatibility (1.44s)
    --- FAIL: TestCompatibility/core.v1.Event (0.02s)
        --- FAIL: TestCompatibility/core.v1.Event/v1.17.0 (0.01s)
            compatibility.go:488: proto differs
            compatibility.go:490:   strings.Join({
                  	... // 79 identical lines
                  	"      2: 456715000",
                  	"    }",
                - 	`    3: ""`,
                  	"  }",
                  	`  12: "30"`,
                  	... // 18 identical lines
                  }, "\n")

            compatibility.go:506: wrote expected compatibility data... verify, commit, and rerun tests
        --- FAIL: TestCompatibility/core.v1.Event/v1.18.0 (0.01s)
            compatibility.go:488: proto differs
            compatibility.go:490:   strings.Join({
                  	... // 79 identical lines
                  	"      2: 456715000",
                  	"    }",
                - 	`    3: ""`,
                  	"  }",
                  	`  12: "30"`,
                  	... // 18 identical lines
                  }, "\n")

            compatibility.go:506: wrote expected compatibility data... verify, commit, and rerun tests
    --- FAIL: TestCompatibility/events.k8s.io.v1beta1.Event (0.02s)
        --- FAIL: TestCompatibility/events.k8s.io.v1beta1.Event/v1.17.0 (0.01s)
            compatibility.go:476: json differs
            compatibility.go:477:   strings.Join({
                  	... // 43 identical lines
                  	`  "series": {`,
                  	`    "count": -1971381490,`,
                - 	`    "lastObservedTime": "2429-02-18T17:57:56.343118Z",`,
                - 	`    "state": "鯹)晿\u003c"`,
                + 	`    "lastObservedTime": "2429-02-18T17:57:56.343118Z"`,
                  	"  },",
                  	`  "reportingController": "19",`,
                  	... // 31 identical lines
                  }, "\n")

            compatibility.go:482: yaml differs
            compatibility.go:483:   strings.Join({
                  	... // 60 identical lines
                  	"  count: -1971381490",
                  	`  lastObservedTime: "2429-02-18T17:57:56.343118Z"`,
                - 	"  state: 鯹)晿<",
                  	`type: "36"`,
                  	"",
                  }, "\n")

            compatibility.go:488: proto differs
            compatibility.go:490:   strings.Join({
                  	... // 54 identical lines
                  	"      2: 343118000",
                  	"    }",
                - 	`    3: "\351\257\271)\346\231\277<"`,
                  	"  }",
                  	`  4: "19"`,
                  	... // 43 identical lines
                  }, "\n")

            compatibility.go:506: wrote expected compatibility data... verify, commit, and rerun tests
        --- FAIL: TestCompatibility/events.k8s.io.v1beta1.Event/v1.18.0 (0.01s)
            compatibility.go:476: json differs
            compatibility.go:477:   strings.Join({
                  	... // 43 identical lines
                  	`  "series": {`,
                  	`    "count": -1971381490,`,
                - 	`    "lastObservedTime": "2429-02-18T17:57:56.343118Z",`,
                - 	`    "state": "鯹)晿\u003c"`,
                + 	`    "lastObservedTime": "2429-02-18T17:57:56.343118Z"`,
                  	"  },",
                  	`  "reportingController": "19",`,
                  	... // 31 identical lines
                  }, "\n")

            compatibility.go:482: yaml differs
            compatibility.go:483:   strings.Join({
                  	... // 60 identical lines
                  	"  count: -1971381490",
                  	`  lastObservedTime: "2429-02-18T17:57:56.343118Z"`,
                - 	"  state: 鯹)晿<",
                  	`type: "36"`,
                  	"",
                  }, "\n")

            compatibility.go:488: proto differs
            compatibility.go:490:   strings.Join({
                  	... // 54 identical lines
                  	"      2: 343118000",
                  	"    }",
                - 	`    3: "\351\257\271)\346\231\277<"`,
                  	"  }",
                  	`  4: "19"`,
                  	... // 43 identical lines
                  }, "\n")

            compatibility.go:506: wrote expected compatibility data... verify, commit, and rerun tests

liggitt commented:
/lgtm
/approve
